testpython.py

```diff
@@ -0,0 +1,97 @@
+                    if "type" in measurement and not isinstance(measurement["type"], str):
+                        errors.append(f"El 'type' debe ser una cadena en: {measurement}")
+                    if "safeRange" in measurement and not isinstance(measurement.get("safeRange", []), list):
+                        errors.append(f"'safeRange' debe ser una lista en: {measurement}")
+                    if "warningRange" in measurement and not isinstance(measurement.get("warningRange", []), list):
+                        errors.append(f"'warningRange' debe ser una lista en: {measurement}")
+                    if "podUnits" in measurement and not isinstance(measurement["podUnits"], str):
+                        errors.append(f"El 'podUnits' debe ser una cadena en: {measurement}") #esto se puede quitar (json 516)
+                elif not isinstance(measurement, str): 
+                    errors.append("Cada elemento en 'measurements' debe ser un objeto o una cadena (nombre de archivo).")
+
+    
+    if "packets" in data:
+        if not isinstance(data["packets"], list):
+            errors.append("La clave 'packets' debe ser una lista.")
+        else:
+            for packet in data["packets"]:
+                if isinstance(packet, dict):  
+                    required_keys = ["id", "name", "type"]
+                    for key in required_keys:
+                        if key not in packet:
+                            errors.append(f"Falta la clave '{key}' en un objeto de 'packets'.")
+                    if "id" in packet and not isinstance(packet["id"], str):
+                        errors.append(f"El 'id' debe ser una cadena en: {packet}")
+                    if "name" in packet and not isinstance(packet["name"], str):
+                        errors.append(f"El 'name' debe ser una cadena en: {packet}")
+                    if "type" in packet and not isinstance(packet["type"], str):
+                        errors.append(f"El 'type' debe ser una cadena en: {packet}")
+                    if "variables" in packet:
+                        if not isinstance(packet["variables"], list):
+                            errors.append(f"'variables' debe ser una lista en: {packet}")
+                        else:
+                            for variable in packet["variables"]:
+                                if not isinstance(variable, dict):
+                                    errors.append(f"Cada elemento en 'variables' debe ser un objeto en: {packet}")
+                                if "name" not in variable:
+                                    errors.append(f"Falta la clave 'name' en un objeto de 'variables' en: {packet}")
+                elif not isinstance(packet, str):  
+                    errors.append("Cada elemento en 'packets' debe ser un objeto o una cadena (nombre de archivo).")
+
+    return errors
+
+def validate_json_folder(folder_path):
+    
+    for root, _, files in os.walk(folder_path):
```

Comment:
Get the folders by knowing the boards listed in `JSON_ADE/boards.json`, the structure from there is `JSON_ADE/boards/board_name/...`